frappe/integrations/doctype/social_login_key/social_login_key.py

```diff
@@ -206,6 +209,20 @@ def get_social_login_provider(self, provider, initialize=False):
 			"auth_url_data": json.dumps({"response_type": "code", "scope": "openid"}),
 		}
 
+		providers["Keycloak"] = {
+			"provider_name": "Keycloak",
+			"enable_social_login": 1,
+			"base_url": "http://localhost:8080/realms/master",
```

Comment:
Don't hardcode something like this - if this is something self-hosted, you can follow Frappe's setup

Reply:
We used it just to test locally. I have updated the code in the latest commit